cg/meta/clean/clean_flow_cells.py

```diff
@@ -0,0 +1,178 @@
+from cg.meta.demultiplex.housekeeper_storage_functions import (
+    get_sample_sheets_from_latest_version,
+)
+from cg.models.demultiplex.flow_cell import FlowCellDirectoryData
+from cg.store import Store
+from cg.store.models import Flowcell, SampleLaneSequencingMetrics
+from cg.utils.files import remove_directory_and_contents
+from cg.utils.time import is_directory_older_than_days_old
+
+LOG = logging.getLogger(__name__)
+
+
+class CleanFlowCellAPI:
+    """
+            Handles the cleaning of flow cells in the flow cells and demultiplexed runs directories.
+    Requirements for cleaning:
+            Flow cell is older than 21 days
+            Flow cell is backed up
+            Flow cell is in StatusDB
+            Flow cell has sequencing metrics in StatusDB
+            Flow cell has fastq files in Housekeeper
+            Flow cell has SPRING files in Housekeeper
+            Flow cell has SPRING metadata in Housekeeper
+            Flow cell has a sample sheet in Housekeeper
+    """
+
+    def __init__(
+        self, status_db: Store, housekeeper_api: HousekeeperAPI, flow_cell_path: Path, dry_run: bool
+    ):
+        self.status_db: Store = status_db
+        self.hk_api: HousekeeperAPI = housekeeper_api
+        self.flow_cell = FlowCellDirectoryData(flow_cell_path=flow_cell_path)
+        self.current_time: float = time.time()
+        self.dry_run: bool = dry_run
+
+    def delete_flow_cell_directory(self) -> bool:
+        """
+        Delete the flow cell directory if it fulfills all requirements.
+        """
+        is_error_raised: bool = False
+        try:
+            self.set_sample_sheet_path_from_housekeeper()
+            if self.can_flow_cell_directory_be_deleted():
+                if self.dry_run:
+                    LOG.debug(f"Dry run: Would have removed: {self.flow_cell.path}")
+                    return is_error_raised
+                remove_directory_and_contents(self.flow_cell.path)
+                return is_error_raised
+        except Exception as error:
+            is_error_raised = True
+            LOG.error(f"Flow cell with path {self.flow_cell.path} not removed: {str(error)}")
+        return is_error_raised
+
+    def set_sample_sheet_path_from_housekeeper(self):
+        """Set the sample sheet for a flow cell."""
+        sample_sheets: Optional[List[File]] = get_sample_sheets_from_latest_version(
+            flow_cell_id=self.flow_cell.id, hk_api=self.hk_api
+        )
+        if not sample_sheets:
+            raise HousekeeperFileMissingError(
+                f"No sample sheet found for flow cell {self.flow_cell.id} in Housekeeper."
+            )
+        sample_sheet_path: Path = Path(
+            get_sample_sheets_from_latest_version(
+                flow_cell_id=self.flow_cell.id, hk_api=self.hk_api
+            )[0].full_path
+        )
+        self.flow_cell.set_sample_sheet_path_hk(sample_sheet_path)
+
+    def can_flow_cell_directory_be_deleted(self) -> bool:
```

Comment:
This is just a thought: This function looks very cleanly implemented but I am wondering if is easily debuggable. I am comparing it with [`is_demultiplex_possible`](https://github.com/Clinical-Genomics/cg/blob/147fb4f81cb6eaa549e7550b2118ec2bef7482bc/cg/apps/demultiplex/demultiplex_api.py#L152) that, I have personally used when something fails, and is super easy to spot which condition failed. It is implemented in a much more verbose way as it checks individually for each condition, but that helps to know where the execution stopped exactly. Would you consider implementing this function as `is_demultiplexing_possible`?